manifests/filter/open_tracing.pp

```diff
@@ -0,0 +1,182 @@
+# == Resource:  repose::filter::open_tracing
+#
+# This is a resource for enabling emitting traces to an OpenTracing tracer (current supports Jaeger).
+# Only usable in >= 8.8.3.0
+#
+# === Parameters
+#
+# [*ensure*]
+# Bool. Ensure config file present/absent
+# Defaults to <tt>present</tt>
+#
+# [*filename*]
+# String. Config filename.
+# Defaults to <tt>open-tracing.cfg.xml</tt>
+#
+# [*service_name*]
+# String. Application name
+# Defaults to <tt>repose</tt>
+#
+# [*connection_type*]
+# String that sets up opentracing connection.  Can be either http or udp
+# Defaults to <tt>udp</tt>
+#
+# [*connection_endpoint*]
+# String. Set to http endpoint to push traces to.  Required if using http connection.
+# Defaults to <tt>http://localhost:12682/api/traces</tt>
+#
+# [*connection_username*]
+# String. Set if http endpoint requires username/password authentication.  Must be used in conjunction with password
+# Defaults to <tt>undef</tt>
+#
+# [*connection_password*]
+# String. Set if http endpoint requires username/password authentication.  Must be used in conjunction with username
+# Defaults to <tt>undef</tt>
+#
+# [*connection_token*]
+# String. Set if http endpoint requires token authentication.
+# Defaults to <tt>undef</tt>
+#
+# [*connection_port*]
+# Integer. Set to port the agent is listening on.  Required if using udp connection.
+# Defaults to <tt>5775</tt>
+#
+# [*connection_host*]
+# String. Set to host the agent is listening on.  Required if using udp connection.
+# Defaults to <tt>localhost</tt>
+#
+# [*sampling_type*]
+# String that sets up opentracing sampling algorithm.  Can be constant, probabilistic, or rate-limiting
+# Defaults to <tt>constant</tt>
+#
+# [*constant_toggle*]
+# String that sets constant sampling. Can be 'on' or 'off'
+# Defaults to <tt>off</tt>
+#
+# [*max_traces_per_second*]
+# Double. Set to the maximum traces to sample per second.  If not defined, uses default of 1.0
+# Defaults to <tt>1.0</tt>
+#
+# [*probability*]
+# Double. Set to probability sampling rate.  Ranges from 0.0 to 1.0.  Defaults to 0.001, which is equivalent to 1 trace for every 1000 being sampled.
+# Defaults to <tt>0.001</tt>
+#
+# === Links
+#
+# * http://www.openrepose.org/versions/latest/services/open-tracing.html
+#
+# === Examples
+#
+# repsoe::filter::open_tracing {
+#  'open_tracing':
+#     service_name => 'awesome-repose',
+#     connection_type => 'http',
+#     connection_endpoint => 'https://jaeger-collector.example.com/api/traces',
+#     sampling_type => 'constant',
+#     constant_toggle => 'on'
+# }
+#
+# === Authors
+#
+# * Dimitry Ushakov <mailto:dimitry-ushakov@rackspace.com>
+#
+define repose::filter::open_tracing (
+  $ensure                   = present,
+  $filename                 = 'open-tracing.cfg.xml',
+  $service_name             = 'repose',
+  $connection_type          = 'udp',
+  $connection_endpoint      = 'http://localhost:12682/api/traces',
```

Comment:
these defaults don't agree. you've specified a default of udp, then provided an endpoint for http.